### PR TITLE
Support for PHP 7.2

### PR DESCRIPTION
**Pull request by StudioMaX**
Reopened https://github.com/mautic/mautic/pull/6599

---

**Please be sure you are submitting this against the _staging_ branch.**

[//]: # ( Please answer the following questions: )

| Q  | A
| --- | ---
| Bug fix? | Y
| New feature? | N
| Automated tests included? | Already covered
| Related user documentation PR URL | /
| Related developer documentation PR URL | /
| Issues addressed (#s or URLs) | #5769 (closed)
| Related PR | #5936
| BC breaks? | N
| Deprecations? | Y

#### Description:
Since there are no fixes for a long time in [original PR](https://github.com/mautic/mautic/pull/5936) to fix the compatibility with PHP 7.2 and because Travis still reports the existing errors, I decided to open a new Pull Request.

It's about time to prepare Mautic for PHP 7.2. This PR upgrades the maximum PHP contstants and fixes PHP warning about Mcrypt constants.

# WARNING
**If you have some plugin credentials stored by Mcrypt they cannot be decrypted by OpenSSL. Make sure you have both libraries installed and enabled. Then re-save all plugins so the credentials are encrypted by OpenSSL and then you can upgrade to PHP 7.2.**

#### Steps to reproduce the bug:
1. Upgrade your PHP to 7.2.
2. You'll see a message that this version is not supported.
3. Hack `MAUTIC_MAXIMUM_PHP` in VersionCheckMiddleware.php to `7.2.999` to overcome the message.
4. Clear the cache.
5. In dev mode you'll see warning everywhere.

#### Steps to test this PR:
1. Apply this PR
2. No warnings are visible. Everything looks normal and little faster :)

#### List deprecations along with the new alternative:
1. Mcrypt extension does not exist on PHP 7.2. Use OpenSSL instead.

### CODE REVIEW

**Comment by Woeler** [comment]
Thanks. Lets wait for Travis now. There are some things that need to be fixed there. Once that is done I will merge this.

**Comment by Woeler** [comment]
@StudioMaX Did you change anything apart from merging? The tests seem to have passed automagically?

**Comment by StudioMaX** [comment]
@Woeler No, I didn't change anything.

File: app/bundles/CoreBundle/Monolog/Handler/PhpHandler.php [hunk 1]
```diff
@@ -1,72 +0,0 @@
-<?php
```
Comment (alanhartless) [inline app/bundles/CoreBundle/Monolog/Handler/PhpHandler.php]: Why was this removed? Unless monolog has changed upstream to write die statements at the beginning of the log, removing this will expose Mautic's logs to the web.

Comment (StudioMaX) [reply app/bundles/CoreBundle/Monolog/Handler/PhpHandler.php]: Please look here: https://github.com/mautic/mautic/pull/6599#discussion-diff-223220882L816

Comment (alanhartless) [reply app/bundles/CoreBundle/Monolog/Handler/PhpHandler.php]: I'm having a hard time finding that discussion. The link takes me to a place that doesn't seem to be the same context. I tried searching the code diff for discussions and couldn't find anything for this context.

Comment (StudioMaX) [reply app/bundles/CoreBundle/Monolog/Handler/PhpHandler.php]: This change is also related: https://github.com/mautic/mautic/pull/6848/files#diff-966ddd047b8bdaf985231305e091f415L816

Why this class was removed:
1. Monolog 2.x previously used constants like `monolog.handler.stream.class` and any bundle was able to redefine it.
2. There was some issues with Monolog 2.x, and it was upgraded to 3.x
3. Monolog 3.x is not supports constant `monolog.handler.stream.class` anymore, and this class cannot be redefined.
4. But actually this file wasn't used since this PR: https://github.com/mautic/mautic/pull/721 when Logger's `type` was changed from `stream` to `rotating_file`, so we can just remove it.

Comment (alanhartless) [reply app/bundles/CoreBundle/Monolog/Handler/PhpHandler.php]: Thanks for that! Hmmm, then that's been an issue for a long time and not good. Will deal with it outside this PR then.

**Comment by Woeler** [comment]
Okay here we go fingers-crossed

**Comment by Woeler** [comment]
I dont know how, but once merged, travis has massive problems https://travis-ci.org/mautic/mautic/builds/452457267?utm_source=github_status&utm_medium=notification

**Comment by StudioMaX** [comment]
Maybe there are some differences between webhooks "continuous-integration/travis-ci/push" and "continuous-integration/travis-ci/pr"?

**Comment by StudioMaX** [comment]
As a workaround we can try to change
https://github.com/mautic/mautic/blob/c445e8c9a0954d561150c250298530af79a43367/app/AppKernel.php#L85-L87
to something like this
```php
        if ((strpos($request->getRequestUri(), 'installer') !== false || !$this->isInstalled())
            && !defined('MAUTIC_INSTALLER')
        ) {
            define('MAUTIC_INSTALLER', 1);
        }
```

But most likely this is because of functional tests in a test environment, when several different tests emulate http-requests via single PHP process.

**Comment by Woeler** [comment]
I am just so confused right now.

**Comment by StudioMaX** [comment]
@Woeler could you test this branch again on your local machine? On my local machine the tests are completed successfully.

**Comment by alanhartless** [comment]
@StudioMaX can you submit this as a new PR since it was reverted via https://github.com/mautic/mautic/pull/6854? 

Try to rebase it to staging `git rebase staging` before creating the PR.
